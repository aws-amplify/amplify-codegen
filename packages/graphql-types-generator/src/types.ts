export type Target = 'json' | 'swift' | 'ts' | 'typescript' | 'flow' | 'scala' | 'flow-modern' | 'angular';
