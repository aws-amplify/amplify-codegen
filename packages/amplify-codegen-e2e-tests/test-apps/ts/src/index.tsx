import React from 'react';
import ReactDOM from 'react-dom/client';
import { Amplify } from 'aws-amplify';
import App from './App';
import awsconfig from './aws-exports';

Amplify.configure(awsconfig);
const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
);
