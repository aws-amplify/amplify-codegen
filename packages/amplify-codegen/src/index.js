const generateTypes = require('./commands/types');
const generateStatements = require('./commands/statements');
const generate = require('./commands/generateStatementsAndType');
const add = require('./commands/add');
const remove = require('./commands/remove');
const configure = require('./commands/configure');
const generateModels = require('./commands/models');
const { generateModelIntrospection, getModelIntrospection } = require('./commands/model-intropection');
const { isCodegenConfigured, switchToSDLSchema } = require('./utils');
const prePushAddGraphQLCodegenHook = require('./callbacks/prePushAddCallback');
const prePushUpdateGraphQLCodegenHook = require('./callbacks/prePushUpdateCallback');
const postPushGraphQLCodegenHook = require('./callbacks/postPushCallback');
const { executeAmplifyCommand, handleAmplifyEvent } = require('./amplify-plugin-index');
const { getCodegenConfig } = require('./codegen-config');

module.exports = {
  configure,
  generate,
  generateTypes,
  generateStatements,
  add,
  remove,
  prePushAddGraphQLCodegenHook,
  prePushUpdateGraphQLCodegenHook,
  postPushGraphQLCodegenHook,
  isCodegenConfigured,
  switchToSDLSchema,
  executeAmplifyCommand,
  handleAmplifyEvent,
  generateModels,
  generateModelIntrospection,
  getModelIntrospection,
  getCodegenConfig,
};

// temp change to trigger release
