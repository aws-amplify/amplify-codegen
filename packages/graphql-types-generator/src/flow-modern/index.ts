export { generateSource, FlowGeneratedFile } from './codeGeneration';
