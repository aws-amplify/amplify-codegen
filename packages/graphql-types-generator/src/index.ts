import './polyfills';

export { default as generate, generateFromString } from './generate';
export type { Target } from './types';
export { getOutputFileName } from './utilities/getOutputFileName';
export { extractDocumentFromJavascript } from './loading';
