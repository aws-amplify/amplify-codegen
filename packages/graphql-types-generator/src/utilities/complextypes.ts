import { GraphQLType, isInputObjectType, getNamedType, isObjectType, isNonNullType } from 'graphql';

// These fields are required for AWSAppsync iOS SDK to create/update
// S3 objects referenced in the API.
const S3_FIELD_NAMES = ['bucket', 'key', 'region', 'localUri', 'mimeType'];

export function hasS3Fields(input: GraphQLType): boolean {
  if (isObjectType(input) || isInputObjectType(input)) {
    if (isS3Field(input)) {
      return true;
    }
    const fields = input.getFields();
    return Object.keys(fields).some((f) => hasS3Fields((<any>fields[f]) as GraphQLType));
  }
  return false;
}

export function isS3Field(field: GraphQLType): boolean {
  if (isObjectType(field) || isInputObjectType(field)) {
    const fields = field.getFields();
    const stringFields = Object.keys(fields).filter((f) => {
      const fieldType = fields[f].type;
      const typeName = getNamedType(fieldType);
      return typeName.name === 'String' && isNonNullType(fieldType);
    });
    const isS3FileField = S3_FIELD_NAMES.every((fieldName) => stringFields.includes(fieldName));
    if (isS3FileField) {
      return true;
    }
  }
  return false;
}
