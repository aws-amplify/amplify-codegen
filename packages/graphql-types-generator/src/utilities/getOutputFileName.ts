import * as path from 'path';
import type { Target } from '../types';

export function getOutputFileName(inputFileName: string, target: Target): string {

  inputFileName = inputFileName || (target === 'angular' ? 'api' : 'API');
  if (supportedTarget(target)) {
    const fileExtension = extensionMap[target];
    const ext = path.extname(inputFileName);
    const baseName = inputFileName.substr(0, inputFileName.length - ext.length);
    const filename = inputFileName.includes(fileExtension) ? inputFileName : `${baseName}.${fileExtension}`;
    return ['API', 'api'].includes(inputFileName) ? path.join(folderMap[target], filename) : filename;
  }
  return inputFileName;
}

const extensionMap: { 'typescript': string, 'flow': string, 'angular': string, 'swift': string } = {
  typescript: 'ts',
  flow: 'js',
  angular: 'service.ts',
  swift: 'swift',
};

const folderMap: { 'typescript': string, 'flow': string, 'angular': string, 'swift': string } = {
  typescript: 'src',
  flow: 'src',
  angular: 'src/app',
  swift: '',
};

function supportedTarget(target: Target): target is 'typescript' | 'flow' | 'angular' | 'swift' {
  return Object.keys(extensionMap).includes(target)
}
