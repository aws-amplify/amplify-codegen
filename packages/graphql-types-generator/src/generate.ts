import { GraphQLSchema, DocumentNode, Source } from 'graphql';
import * as fs from 'fs-extra';
import * as path from 'path';
import * as rimraf from 'rimraf';

import { loadSchema, loadAndMergeQueryDocuments, parseSchema, parseAndMergeQueryDocuments } from './loading';
import { validateQueryDocument } from './validation';
import { compileToIR } from './compiler';
import { compileToLegacyIR } from './compiler/legacyIR';
import serializeToJSON from './serializeToJSON';
import { BasicGeneratedFileMap, isBasicGeneratedFileMap } from './utilities/CodeGenerator';
import { generateSource as generateSwiftSource } from './swift';
import { generateSource as generateTypescriptSource } from './typescript';
import { generateSource as generateFlowSource } from './flow';
import { generateSource as generateFlowModernSource } from './flow-modern';
import { generateSource as generateScalaSource } from './scala';
import { generateSource as generateAngularSource } from './angular';
import { hasS3Fields } from './utilities/complextypes';
import { Target } from './types';
import { getOutputFileName } from './utilities/getOutputFileName';

export default function generate(
  inputPaths: string[],
  schemaPath: string,
  outputPath: string,
  only: string,
  target: Target,
  tagName: string,
  options: any,
): void {
  generateFromFile(inputPaths, schemaPath, outputPath, only, target, tagName, options);
}

function generateFromFile(
  inputPaths: string[],
  schemaPath: string,
  outputPath: string,
  only: string,
  target: Target,
  tagName: string,
  options: any,
) {
  const schema = loadSchema(schemaPath);

  const document = loadAndMergeQueryDocuments(inputPaths, tagName);

  validateQueryDocument(schema, document);
  const multipleFiles = fs.existsSync(outputPath) && fs.statSync(outputPath).isDirectory();

  const output = generateForTarget(schema, document, only, target, multipleFiles, options);

  if (outputPath) {
    if (target === 'flow-modern') {
      // Group by output directory
      const filesByOutputDirectory: {
        [outputDirectory: string]: BasicGeneratedFileMap;
      } = {};

      // if target is flow-modern output must be BasicGeneratedFileMap
      Object.entries(output as BasicGeneratedFileMap).forEach(([filePath, file]) => {
        const outputDirectory = path.dirname(filePath);
        if (!filesByOutputDirectory[outputDirectory]) {
          filesByOutputDirectory[outputDirectory] = {
            [path.basename(filePath)]: file,
          };
        } else {
          filesByOutputDirectory[outputDirectory][path.basename(filePath)] = file;
        }
      });

      Object.keys(filesByOutputDirectory).forEach(outputDirectory => {
        writeGeneratedFiles(filesByOutputDirectory[outputDirectory], outputDirectory);
      });
    } else if (isBasicGeneratedFileMap(output)) {
      writeGeneratedFiles(output, outputPath);
    } else {
      fs.outputFileSync(outputPath, output);
    }
  } else {
    console.log(output);
  }
}

export function generateFromString(
  schema: string,
  introspection: boolean,
  queryDocuments: string,
  target: Target,
  multipleSwiftFiles: boolean,
  options: any,
): { [filepath: string]: string } {
  const graphqlSchema = parseSchema(schema, introspection);
  const document = parseAndMergeQueryDocuments([new Source(queryDocuments)]);
  validateQueryDocument(graphqlSchema, document);
  const output = generateForTarget(graphqlSchema, document, '', target, multipleSwiftFiles, options);

  if (isBasicGeneratedFileMap(output)) {
    return Object.entries(output)
      .map(([filepath, file]) => [filepath, file.output])
      .reduce((acc, [filepath, fileOutput]) => ({ ...acc, [filepath]: fileOutput }), {});
  }

  const filename = getOutputFileName('', target);
  return { [filename]: output };
}

export function generateForTarget(
  schema: GraphQLSchema,
  document: DocumentNode,
  only: string,
  target: Target,
  multipleFiles: boolean,
  options: any,
): string | BasicGeneratedFileMap {
  if (target === 'swift') {
    return generateTypesSwift(schema, document, only, multipleFiles, options);
  }
  if (target === 'flow-modern') {
    return generateTypesFlowModern(schema, document, options);
  }

  const context = compileToLegacyIR(schema, document, options);

  switch (target) {
    case 'json':
      return serializeToJSON(context);
    case 'ts':
    case 'typescript':
      return generateTypescriptSource(context);
    case 'flow':
      return generateFlowSource(context);
    case 'scala':
      return generateScalaSource(context, options);
    case 'angular':
      return generateAngularSource(context);
    default:
      throw new Error(`${target} is not supported.`);
  }
}

function generateTypesSwift(
  schema: GraphQLSchema,
  document: DocumentNode,
  only: string,
  multipleFiles: boolean,
  options: any,
): string | BasicGeneratedFileMap {
  options.addTypename = true;
  const context = compileToIR(schema, document, options);
  // Complex object suppport
  if (options.complexObjectSupport === 'auto') {
    options.addS3Wrapper = context.typesUsed.some(typesUsed => hasS3Fields(typesUsed));
  } else if (options.complexObjectSupport === 'yes') {
    options.addS3Wrapper = true;
  } else {
    options.addS3Wrapper = false;
  }

  const generator = generateSwiftSource(context, multipleFiles, only);

  if (multipleFiles) {
    return generator.generatedFiles;
  }
  return generator.output;
}

function generateTypesFlowModern(schema: GraphQLSchema, document: DocumentNode, options: any): BasicGeneratedFileMap {
  const context = compileToIR(schema, document, options);
  const generatedFiles = generateFlowModernSource(context);

  return generatedFiles;
}

function writeGeneratedFiles(generatedFiles: BasicGeneratedFileMap, outputDirectory: string) {
  // Clear all generated stuff to make sure there isn't anything
  // unnecessary lying around.
  rimraf.sync(outputDirectory);
  // Remake the output directory
  fs.mkdirSync(outputDirectory);

  for (const [fileName, generatedFile] of Object.entries(generatedFiles)) {
    fs.writeFileSync(path.join(outputDirectory, fileName), generatedFile.output);
  }
}
