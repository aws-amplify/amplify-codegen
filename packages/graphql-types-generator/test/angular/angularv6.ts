import { CodeGenerator } from '../../src/utilities/CodeGenerator';
import { compileToLegacyIR } from '../../src/compiler/legacyIR';
import { parse } from 'graphql';
import { generateSource } from '../../src/angular';
import { loadSchema } from '../../src/loading';

const starWarsSchema = loadSchema(require.resolve('../fixtures/starwars/schema.json'));
const subscriptionSchema = loadSchema(require.resolve('../fixtures/misc/subscriptionSchema.json'));
const subscriptionSchemaWithParameters = require.resolve('../fixtures/misc/subscriptionSchemaWithParameters.graphql')

describe('Angular code generation', () => {
  let generator;
  let compileFromSource;
  let addFragment;

  const setup = (schema) => {
    const context = {
      schema: schema,
      operations: {},
      fragments: {},
      typesUsed: {},
    };

    generator = new CodeGenerator(context);

    compileFromSource = source => {
      const document = parse(source);
      const context = compileToLegacyIR(schema, document, {
        mergeInFieldsFromFragmentSpreads: true,
        addTypename: true,
      });
      generator.context = context;
      return context;
    };

    addFragment = fragment => {
      generator.context.fragments[fragment.fragmentName] = fragment;
    };

    return { generator, compileFromSource, addFragment };
  }

  const generateAngularV6API = (context) => generateSource(context, { isAngularV6: true })

  test(`should generate simple query operations`, function() {
    const { compileFromSource } = setup(starWarsSchema);
    const context = compileFromSource(`
      query HeroName {
        hero {
          name
        }
      }
    `);

    const source = generateAngularV6API(context);
    expect(source).toMatchSnapshot();
  });

  test(`should generate simple query operations including input variables`, function() {
    const { compileFromSource } = setup(starWarsSchema);
    const context = compileFromSource(`
      query HeroName($episode: Episode) {
        hero(episode: $episode) {
          name
        }
      }
    `);

    const source = generateAngularV6API(context);
    expect(source).toMatchSnapshot();
  });

  test(`should generate subscriptions`, function() {
    const { compileFromSource } = setup(subscriptionSchema);
    const context = compileFromSource(`
    subscription OnCreateRestaurant {
      onCreateRestaurant {
        id
        name
        description
        city
      }
    }
    `);

    const source = generateAngularV6API(context);
    expect(source).toMatchSnapshot();
  });

  test('should generate subscriptions with parameters', () => {
    const { compileFromSource } = setup(loadSchema(subscriptionSchemaWithParameters));
    const context = compileFromSource(`
    subscription OnCreateRestaurant($owner: String!) {
      onCreateRestaurant(owner: $owner) {
        id
        name
        description
        city
        owner
        createdAt
        updatedAt
      }
    }
    `);
    const source = generateAngularV6API(context);
    expect(source).toMatchSnapshot();
  })
})