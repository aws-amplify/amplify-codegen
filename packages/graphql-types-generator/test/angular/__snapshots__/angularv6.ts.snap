// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Angular code generation should generate simple query operations 1`] = `
"/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.
import { Injectable } from \\"@angular/core\\";
import { Client, generateClient, GraphQLResult } from \\"aws-amplify/api\\";
import { Observable } from \\"rxjs\\";

export type Character = {
  __typename: \\"Character\\";
  // The ID of the character
  id: string;
  // The name of the character
  name: string;
  // The friends of the character, or an empty list if they have none
  friends?: Array<Character | null> | null;
  // The friends of the character exposed as a connection with edges
  friendsConnection: FriendsConnection;
  // The movies this character appears in
  appearsIn: Array<Episode | null>;
};

export type Human = {
  __typename: \\"Human\\";
  // The ID of the human
  id: string;
  // What this human calls themselves
  name: string;
  // The home planet of the human, or null if unknown
  homePlanet?: string | null;
  // Height in the preferred unit, default is meters
  height?: number | null;
  // Mass in kilograms, or null if unknown
  mass?: number | null;
  // This human's friends, or an empty list if they have none
  friends?: Array<Character | null> | null;
  // The friends of the human exposed as a connection with edges
  friendsConnection: FriendsConnection;
  // The movies this human appears in
  appearsIn: Array<Episode | null>;
  // A list of starships this person has piloted, or an empty list if none
  starships?: Array<Starship | null> | null;
};

export type FriendsConnection = {
  __typename: \\"FriendsConnection\\";
  // The total number of friends
  totalCount?: number | null;
  // The edges for each of the character's friends.
  edges?: Array<FriendsEdge | null> | null;
  // A list of the friends, as a convenience when edges are not needed.
  friends?: Array<Character | null> | null;
  // Information for paginating this connection
  pageInfo: PageInfo;
};

export type FriendsEdge = {
  __typename: \\"FriendsEdge\\";
  // A cursor used for pagination
  cursor: string;
  // The character represented by this friendship edge
  node?: Character | null;
};

export type PageInfo = {
  __typename: \\"PageInfo\\";
  startCursor?: string | null;
  endCursor?: string | null;
  hasNextPage: boolean;
};

// The episodes in the Star Wars trilogy
export enum Episode {
  NEWHOPE = \\"NEWHOPE\\", // Star Wars Episode IV: A New Hope, released in 1977.
  EMPIRE = \\"EMPIRE\\", // Star Wars Episode V: The Empire Strikes Back, released in 1980.
  JEDI = \\"JEDI\\" // Star Wars Episode VI: Return of the Jedi, released in 1983.
}

export type Starship = {
  __typename: \\"Starship\\";
  // The ID of the starship
  id: string;
  // The name of the starship
  name: string;
  // Length of the starship, along the longest axis
  length?: number | null;
  coordinates?: Array<Array<number>> | null;
};

export type Droid = {
  __typename: \\"Droid\\";
  // The ID of the droid
  id: string;
  // What others call this droid
  name: string;
  // This droid's friends, or an empty list if they have none
  friends?: Array<Character | null> | null;
  // The friends of the droid exposed as a connection with edges
  friendsConnection: FriendsConnection;
  // The movies this droid appears in
  appearsIn: Array<Episode | null>;
  // This droid's primary function
  primaryFunction?: string | null;
};

export type HeroNameQuery = {
  __typename: \\"Character\\";
  // The name of the character
  name: string;
};

@Injectable({
  providedIn: \\"root\\"
})
export class APIService {
  public client: Client;
  constructor() {
    this.client = generateClient();
  }
  async HeroName(): Promise<HeroNameQuery> {
    const statement = \`query HeroName {
        hero {
          __typename
          name
        }
      }\`;
    const response = (await this.client.graphql({ query: statement })) as any;
    return <HeroNameQuery>response.data.hero;
  }
}
"
`;

exports[`Angular code generation should generate simple query operations including input variables 1`] = `
"/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.
import { Injectable } from \\"@angular/core\\";
import { Client, generateClient, GraphQLResult } from \\"aws-amplify/api\\";
import { Observable } from \\"rxjs\\";

// The episodes in the Star Wars trilogy
export enum Episode {
  NEWHOPE = \\"NEWHOPE\\", // Star Wars Episode IV: A New Hope, released in 1977.
  EMPIRE = \\"EMPIRE\\", // Star Wars Episode V: The Empire Strikes Back, released in 1980.
  JEDI = \\"JEDI\\" // Star Wars Episode VI: Return of the Jedi, released in 1983.
}

export type Character = {
  __typename: \\"Character\\";
  // The ID of the character
  id: string;
  // The name of the character
  name: string;
  // The friends of the character, or an empty list if they have none
  friends?: Array<Character | null> | null;
  // The friends of the character exposed as a connection with edges
  friendsConnection: FriendsConnection;
  // The movies this character appears in
  appearsIn: Array<Episode | null>;
};

export type Human = {
  __typename: \\"Human\\";
  // The ID of the human
  id: string;
  // What this human calls themselves
  name: string;
  // The home planet of the human, or null if unknown
  homePlanet?: string | null;
  // Height in the preferred unit, default is meters
  height?: number | null;
  // Mass in kilograms, or null if unknown
  mass?: number | null;
  // This human's friends, or an empty list if they have none
  friends?: Array<Character | null> | null;
  // The friends of the human exposed as a connection with edges
  friendsConnection: FriendsConnection;
  // The movies this human appears in
  appearsIn: Array<Episode | null>;
  // A list of starships this person has piloted, or an empty list if none
  starships?: Array<Starship | null> | null;
};

export type FriendsConnection = {
  __typename: \\"FriendsConnection\\";
  // The total number of friends
  totalCount?: number | null;
  // The edges for each of the character's friends.
  edges?: Array<FriendsEdge | null> | null;
  // A list of the friends, as a convenience when edges are not needed.
  friends?: Array<Character | null> | null;
  // Information for paginating this connection
  pageInfo: PageInfo;
};

export type FriendsEdge = {
  __typename: \\"FriendsEdge\\";
  // A cursor used for pagination
  cursor: string;
  // The character represented by this friendship edge
  node?: Character | null;
};

export type PageInfo = {
  __typename: \\"PageInfo\\";
  startCursor?: string | null;
  endCursor?: string | null;
  hasNextPage: boolean;
};

export type Starship = {
  __typename: \\"Starship\\";
  // The ID of the starship
  id: string;
  // The name of the starship
  name: string;
  // Length of the starship, along the longest axis
  length?: number | null;
  coordinates?: Array<Array<number>> | null;
};

export type Droid = {
  __typename: \\"Droid\\";
  // The ID of the droid
  id: string;
  // What others call this droid
  name: string;
  // This droid's friends, or an empty list if they have none
  friends?: Array<Character | null> | null;
  // The friends of the droid exposed as a connection with edges
  friendsConnection: FriendsConnection;
  // The movies this droid appears in
  appearsIn: Array<Episode | null>;
  // This droid's primary function
  primaryFunction?: string | null;
};

export type HeroNameQuery = {
  __typename: \\"Character\\";
  // The name of the character
  name: string;
};

@Injectable({
  providedIn: \\"root\\"
})
export class APIService {
  public client: Client;
  constructor() {
    this.client = generateClient();
  }
  async HeroName(episode?: Episode): Promise<HeroNameQuery> {
    const statement = \`query HeroName($episode: Episode) {
        hero(episode: $episode) {
          __typename
          name
        }
      }\`;
    const gqlAPIServiceArguments: any = {};
    if (episode) {
      gqlAPIServiceArguments.episode = episode;
    }
    const response = (await this.client.graphql({
      query: statement,
      variables: gqlAPIServiceArguments
    })) as any;
    return <HeroNameQuery>response.data.hero;
  }
}
"
`;

exports[`Angular code generation should generate subscriptions 1`] = `
"/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.
import { Injectable } from \\"@angular/core\\";
import { Client, generateClient, GraphQLResult } from \\"aws-amplify/api\\";
import { Observable } from \\"rxjs\\";

export type __SubscriptionContainer = {
  onCreateRestaurant: OnCreateRestaurantSubscription;
};

export type Restaurant = {
  __typename: \\"Restaurant\\";
  id: string;
  name: string;
  description: string;
  city: string;
};

export type OnCreateRestaurantSubscription = {
  __typename: \\"Restaurant\\";
  id: string;
  name: string;
  description: string;
  city: string;
};

@Injectable({
  providedIn: \\"root\\"
})
export class APIService {
  public client: Client;
  constructor() {
    this.client = generateClient();
  }
  OnCreateRestaurantListener(): Observable<
    GraphQLResult<Pick<__SubscriptionContainer, \\"onCreateRestaurant\\">>
  > {
    return this.client.graphql({
      query: \`subscription OnCreateRestaurant {
        onCreateRestaurant {
          __typename
          id
          name
          description
          city
        }
      }\`
    }) as any;
  }
}
"
`;

exports[`Angular code generation should generate subscriptions with parameters 1`] = `
"/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.
import { Injectable } from \\"@angular/core\\";
import { Client, generateClient, GraphQLResult } from \\"aws-amplify/api\\";
import { Observable } from \\"rxjs\\";

export type __SubscriptionContainer = {
  onCreateRestaurant: OnCreateRestaurantSubscription;
};

export type Restaurant = {
  __typename: \\"Restaurant\\";
  id: string;
  name: string;
  description: string;
  city: string;
  owner?: string | null;
  createdAt: string;
  updatedAt: string;
};

export type OnCreateRestaurantSubscription = {
  __typename: \\"Restaurant\\";
  id: string;
  name: string;
  description: string;
  city: string;
  owner?: string | null;
  createdAt: string;
  updatedAt: string;
};

@Injectable({
  providedIn: \\"root\\"
})
export class APIService {
  public client: Client;
  constructor() {
    this.client = generateClient();
  }
  OnCreateRestaurantListener(
    owner: string
  ): Observable<
    GraphQLResult<Pick<__SubscriptionContainer, \\"onCreateRestaurant\\">>
  > {
    const statement = \`subscription OnCreateRestaurant($owner: String!) {
        onCreateRestaurant(owner: $owner) {
          __typename
          id
          name
          description
          city
          owner
          createdAt
          updatedAt
        }
      }\`;
    const gqlAPIServiceArguments: any = {
      owner
    };
    return this.client.graphql({
      query: statement,
      variables: gqlAPIServiceArguments
    }) as any;
  }
}
"
`;
